service/security/shiro/src/main/java/org/eclipse/kapua/service/authentication/shiro/realm/UserPassCredentialsMatcher.java

```diff
@@ -178,4 +187,18 @@ public boolean doCredentialsMatch(AuthenticationToken authenticationToken, Authe
         return credentialMatch;
     }
 
+    private boolean checkPassword(String tokenPassword, Credential infoCredential) {
+        String cachedCredential = CACHED_CREDENTIALS.get(tokenPassword);
```

Comment:
Perhaps we mught want to do at lease a `SHA-xxx` on the key of the map so we don't have clear-text passwords stored in memory?

Reply:
done, please see my changes